### ISSUE TITLE
IntelliSense support for Bindings in XAML

### ISSUE THREAD

**Issue by MAUIoxo** [dotnet/maui#26077]
### Description

Would be good to have IntelliSense support while designing UI‘s in XAML to set Bindings of properties more easily.

A use cases would be showing all possible properties that can be bound in an IntelliSense menu, e.g. from the `ViewModel` of the current `BindingContext`.

This can also be enhanced to more complex scenarios when there is a need to reference some parts from other `ViewModels` like in the following simple example:

```XAML
<ListView ItemsSource="{Binding Items}">
    <ListView.ItemTemplate>
        <DataTemplate>
            <StackPanel>
                <!-- Bindung of a Property from the bound  ItemViewModel -->
                <TextBlock Text="{Binding ItemName}" />
                
                <!-- Bindung of a Property of a different MainViewModel -->
                <TextBlock Text="{Binding DataContext.MainTitle, 
                                          RelativeSource={RelativeSource AncestorType=Window}}" />
            </StackPanel>
        </DataTemplate>
    </ListView.ItemTemplate>
</ListView>
```

Users shouldn’t also have to write and know all possibilities how to write these references like `RelativeSource` etc. and this should all be resolved automatically and correctly via IntelliSense. 

### Public API Changes

Don’t know about API changes

### Intended Use-Case

Easier support for Users/Developers to reference Bindings in XAML across different `ViewModels`. All properties should be listet in an IntelliSense menu and when selected they should be referenced in the XAML code correctly like a necessary `RelativeSource` or `Path` declaration

**Comment by MAUIoxo** [comment]
I think that will work for simple Bindings in the current `BindingContext` but, so far, I can't remember that I have seen IntelliSense so that when selected it writes the whole `RepativeSource`, right?

**Comment by mgoertz-msft** [comment]
No, it doesn't write the whole `RelativeSource` for you. How would it know what you want there?

**Comment by MAUIoxo** [comment]
Well, IntelliSense could list accessible properties from a `ViewModel` or all available properties if would be an easy task to create this for you.

@OvrBtn:  In the following case with an `x:Type` IntelliSense even does not appear when I type "=" behind `Path` and I already specified a lot like `{RelativeSource AncestorType={x:Type viewModels:MainPageViewModel}, Path=}`. Here, it should be possible to list all accessible properties.

In general, I should be possible to select from a list of available properties from the app and after a selection it would be good when it writes the entire source like `IsVisible={RelativeSource AncestorType={x:Type viewModels:MainPageViewModel}, Path="IsItemVisibleProperty"}`
